torch_frame/data/mapper.py

```diff
@@ -129,13 +130,25 @@ def forward(
         if ser.dtype != 'object':
             raise ValueError('Multi-categorical types expect string as input')
         values = []
-        ser = ser.apply(self._row_to_tensor)
-        values = torch.tensor(sum(ser, []), device=device)
-        ser = ser.apply(lambda x: len(x))
-        offset = torch.tensor([0] + ser.tolist(), device=device)
+        original_index = ser.index
+        ser = ser.apply(self._split_by_sep)
+        ser = ser.explode()
+        ser = pd.merge(
+            ser.rename('data'),
+            self.index.rename('index'),
+            how='left',
+            left_on='data',
+            right_index=True,
+        ).dropna()
+        ser['index'] = ser['index'].astype('int64')
+        values = torch.tensor(ser['index'].values, device=device)
+        offset = ser.index.value_counts()
+        offset = offset.reindex(original_index, fill_value=0)
+        offset = pd.concat((pd.Series([0]), offset))
+        offset = torch.tensor(offset.values, device=device)
         offset = torch.cumsum(offset, dim=0)
```

Comment:
Just curious, why you don't need to cast to torch.long here?

Reply:
I think I obtained the `offset` from `ser.index.value_counts()`. So I assume `value_counts()` by default returns int64 type.

Reply:
Ok, I just wanted to confirm since you explicitly casted as `long` before